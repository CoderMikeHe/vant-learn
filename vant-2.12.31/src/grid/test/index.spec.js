import { mount } from '../../../test';

test('click grid item', () => {
  const onClick = jest.fn();
  const wrapper = mount({
    template: `
      <van-grid>
        <van-grid-item @click="onClick" />
      </van-grid>
    `,
    methods: {
      onClick,
    },
  });

  const Item = wrapper.find('.van-grid-item__content');
  Item.trigger('click');

  expect(onClick).toHaveBeenCalledTimes(1);
});

test('square and set gutter', () => {
  const wrapper = mount({
    template: `
      <van-grid square :column-num="2" gutter="10rem">
        <van-grid-item />
        <van-grid-item />
        <van-grid-item />
      </van-grid>
    `,
  });

  expect(wrapper).toMatchSnapshot();
});

test('icon-size prop', () => {
  const wrapper = mount({
    template: `
      <van-grid icon-size="10">
        <van-grid-item icon="success" />
      </van-grid>
    `,
  });

  expect(wrapper).toMatchSnapshot();
});

test('render icon-slot', () => {
  const wrapper = mount({
    template: `
      <van-grid icon-size="10">
        <van-grid-item badge="1">
          <template #icon>
            <div>Custom Icon</div>
          </template>
        </van-grid-item>
      </van-grid>
    `,
  });

  expect(wrapper).toMatchSnapshot();
});
