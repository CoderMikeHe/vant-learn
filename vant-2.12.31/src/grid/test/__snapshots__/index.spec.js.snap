// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon-size prop 1`] = `
<div class="van-grid van-hairline--top">
  <div class="van-grid-item" style="flex-basis: 25%;">
    <div class="van-grid-item__content van-grid-item__content--center van-hairline"><i class="van-icon van-icon-success van-grid-item__icon" style="font-size: 10px;">
        <!----></i></div>
  </div>
</div>
`;

exports[`render icon-slot 1`] = `
<div class="van-grid van-hairline--top">
  <div class="van-grid-item" style="flex-basis: 25%;">
    <div class="van-grid-item__content van-grid-item__content--center van-hairline">
      <div class="van-grid-item__icon-wrapper">
        <div>Custom Icon</div>
        <div class="van-info">1</div>
      </div>
    </div>
  </div>
</div>
`;

exports[`square and set gutter 1`] = `
<div class="van-grid" style="padding-left: 10rem;">
  <div class="van-grid-item van-grid-item--square" style="flex-basis: 50%; padding-top: 50%;">
    <div class="van-grid-item__content van-grid-item__content--center van-grid-item__content--square van-grid-item__content--surround van-hairline" style="right: 10rem; bottom: 10rem; height: auto;"></div>
  </div>
  <div class="van-grid-item van-grid-item--square" style="flex-basis: 50%; padding-top: 50%;">
    <div class="van-grid-item__content van-grid-item__content--center van-grid-item__content--square van-grid-item__content--surround van-hairline" style="right: 10rem; bottom: 10rem; height: auto;"></div>
  </div>
  <div class="van-grid-item van-grid-item--square" style="flex-basis: 50%; padding-top: 50%;">
    <div class="van-grid-item__content van-grid-item__content--center van-grid-item__content--square van-grid-item__content--surround van-hairline" style="right: 10rem; bottom: 10rem; height: auto;"></div>
  </div>
</div>
`;
